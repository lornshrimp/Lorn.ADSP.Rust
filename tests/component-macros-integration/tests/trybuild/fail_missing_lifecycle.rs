use component_macros::{lifecycle};

// Intentionally incorrect: applying lifecycle without required struct definition context
#[lifecycle(on_start = "start")] // expect failure because no struct follows
fn dummy() {}

fn main() {}
