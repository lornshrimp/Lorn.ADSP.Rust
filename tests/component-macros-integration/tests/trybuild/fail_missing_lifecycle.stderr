error: expected `struct`
 --> tests/trybuild/fail_missing_lifecycle.rs:5:1
  |
5 | fn dummy() {}
  | ^^
